### PR TITLE
[Merged by Bors] - fix: Increase default pvc size to 1Gi as required by IONOS

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by maltesander — APPROVED** [review]
LGTM if tests pass (maybe add a changelong entry for the why?)

**Comment by sbernauer** [comment]
I don't think this must be mentioned explicitly in the changelog, as this is IMHO an detail of how we decided on what sizes the pcvs should have by default

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hdfs-operator-it-custom/89/

**Review by maltesander — APPROVED** [review]
LGTM!
Edit: The kerberos tests failed?

File: docs/modules/hdfs/pages/usage-guide/resources.adoc [hunk 1]
```diff
@@ -84,7 +91,7 @@ spec:
           limit: 512Mi
         storage:
           data:
-            capacity: 512Mi
+            capacity: 1Gi
```
Comment (Techassi) [inline docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: I guess we should be consistent here. Most of the resource related PRs use `Mi`. So in this case `1024MI` and `2048Mi`. Same goes for the `Quantity` in `JournalNodeConfigFragment`.

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -1138,7 +1138,7 @@ impl JournalNodeConfigFragment {
                 },
                 storage: HdfsStorageConfigFragment {
                     data: PvcConfigFragment {
-                        capacity: Some(Quantity("512Mi".to_owned())),
+                        capacity: Some(Quantity("1Gi".to_owned())),
```
Comment (Techassi) [inline rust/crd/src/lib.rs:1140]: See above comment

File: docs/modules/hdfs/pages/usage-guide/resources.adoc (hunk 1 above)
Comment (sbernauer) [reply docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: I was aware of this and did it intentionally, as I prefer using Gi over Mi over Ki over writing raw numbers ;)
What will end up in k8s will be 1Gi anyway AFAIK.
But I'm happy to let the majority decide or change it back when you have stronger feelings than I have ;)

Comment (maltesander) [reply docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: Yeah i think 1024Mi will show up as 1Gi. I would like to set what will actually be "returned" by the api server for easier testing? Meaning always transform to the highest unit still keeping the natural number.
Its similar for 1000m and "1" CPU.
But yes we should keep this consistent.

Comment (sbernauer) [reply docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: Let's flip a coin. head or tail?
Jokes aside, would let you decide

Comment (Techassi) [reply docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: For now, I would keep it consistent. Maybe we want to come up with an internal policy (and document it) which syntax / unit we use, and also at what value we switch from one unit to another.

**Comment by sbernauer** [comment]
Hopefully last test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hdfs-operator-it-custom/90

**Comment by sbernauer** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hdfs-operator-it-custom/91/

File: docs/modules/hdfs/pages/usage-guide/resources.adoc (hunk 1 above)
Comment (sbernauer) [reply docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: Wait, are you talking about the pvc size or Memory? I *think* we use Mi or Memory in most operators and Gi for pvc sizes (as every pvc has at a minimum 1Gi)

Comment (sbernauer) [reply docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: Added it to Arch meeting

Comment (sbernauer) [reply docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: Ok, did walk through the code and we actually use `Mi` for all the memory declaration, event when we could have used Gi

Comment (Techassi) [reply docs/modules/hdfs/pages/usage-guide/resources.adoc:94]: Okay that's great! Putting further discussion into the arch meeting seems like a good idea.

**Review by Techassi — APPROVED** [review]

**Comment by sbernauer** [comment]
bors r+
Thanks for review, tests have passed :rocket:

**Pull request closed without merging**